### DISCUSSION TITLE
`CRON_SECRET` is `undefined` when running Vercel Cron

### DISCUSSION THREAD

**Discussion by devjiwonchoi** [vercel/community#7162] [Help] [answered]
### Page to Investigate

https://vercel.com/docs/cron-jobs/manage-cron-jobs#securing-cron-jobs

### Steps to Reproduce

NOTE: I'm running cron on `route.ts` in Next.js.

```ts
export async function GET(request: Request) {
  const authHeader = request.headers.get('authorization')
  if (
    process.env.NODE_ENV === 'production' &&
    authHeader !== `Bearer ${process.env.CRON_SECRET}`
  ) {
    console.log(process.env.CRON_SECRET, authHeader) // undefined null
    return new Response('Unauthorized.', { status: 401 })
  }
}
```

![Screenshot 2024-06-12 at 12 11 54 AM](https://github.com/vercel/community/assets/120007119/9aedff7c-908f-4a19-b6a7-7d82d2879707)

I get `undefined null` from the Vercel Log.

**Comment by versecafe** [comment, marked as answer]
@devjiwonchoi Have you set a `CRON_SECRET` in your projects env vars?

**Reply by devjiwonchoi** [reply]
🤦🏻‍♂️silly me, thought it is sent from Vercel. Thank you!

[a comment by an automated account is not shown]
